src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -39,14 +39,10 @@ public abstract partial class ClientApplicationBase : ApplicationBase, IClientAp
 
         internal ClientApplicationBase(ApplicationConfiguration config) : base(config)
         {
-            if (config.UserTokenLegacyCachePersistenceForTest != null)
-            {
-                UserTokenCacheInternal = new TokenCache(ServiceBundle, config.UserTokenLegacyCachePersistenceForTest, false, defaultCacheSerialization);
-            }
-            else
-            {
-                UserTokenCacheInternal = config.UserTokenCacheInternalForTest ?? new TokenCache(ServiceBundle, false, defaultCacheSerialization);
-            }
+            UserTokenCacheInternal = new TokenCache(
+                ServiceBundle,
+                config.UserTokenLegacyCachePersistenceForTest,
+                isApplicationTokenCache: false);
```

Comment:
`UserTokenCacheInternalForTest` is used here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/eac7c9af728006061874e96066649e4245a36f2f/tests/Microsoft.Identity.Test.Unit/CacheTests/TokenCacheNotificationTests.cs#L368

```suggestion
            UserTokenCacheInternal = config.UserTokenCacheInternalForTest ?? new TokenCache(
                ServiceBundle,
                config.UserTokenLegacyCachePersistenceForTest,
                isApplicationTokenCache: false);
```